next batch of nested testsets

sbromberger opened the pull request.

simonschoelly commented:
I did not spot anything wrong, but then there is also a lot of code. I suppose there is no difference in logic?

sbromberger commented:
> I suppose there is no difference in logic?

There's not really any difference in logic (the codecov is the same), but I moved a few things around and simplified where possible.
